src/main/java/com/zaxxer/hikari/proxy/ConnectionProxy.java

```diff
@@ -160,11 +161,13 @@ private final void closeOpenStatements()
             }
             catch (SQLException e) {
                checkException(e);
+               success = success ?  !poolEntry.evict : false;
             }
          }
 
          openStatements.clear();
       }
+      return success;
```

Comment:
Maybe it wasn't clean, I mean [this](https://gist.github.com/brettwooldridge/22171b1b83f7882d6aa7#file-snippet-java).